internal/build/phases_test.go

```diff
@@ -47,6 +48,166 @@ func testPhases(t *testing.T, when spec.G, it spec.S) {
+			fakePhaseFactory := fakes.NewFakePhaseFactory(fakes.WhichReturnsForNew(fakePhase))
+
+			err := lifecycle.Create(context.Background(), false, false, "test", "test", "test", "test", "test", fakePhaseFactory)
+			h.AssertNil(t, err)
+
+			h.AssertEq(t, fakePhase.CleanupCallCount, 1)
+			h.AssertEq(t, fakePhase.RunCallCount, 1)
+		})
+
+		it("configures the phase with the expected arguments", func() {
+			verboseLifecycle := fakeLifecycle(t, true)
+			fakePhaseFactory := fakes.NewFakePhaseFactory()
+			expectedRepoName := "some-repo-name"
+			expectedRunImage := "some-run-image"
+
+			err := verboseLifecycle.Create(context.Background(), false, false, expectedRunImage, "test", "test", expectedRepoName, "test", fakePhaseFactory)
+			h.AssertNil(t, err)
+
+			configProvider := fakePhaseFactory.NewCalledWithProvider
+			h.AssertEq(t, configProvider.Name(), "creator")
+			h.AssertIncludeAllExpectedPatterns(t,
+				configProvider.ContainerConfig().Cmd,
+				[]string{"-log-level", "debug"},
+				[]string{"-run-image", expectedRunImage},
+				[]string{expectedRepoName},
+			)
+		})
+
+		it("configures the phase with the expected network mode", func() {
+			lifecycle := fakeLifecycle(t, false)
+			fakePhaseFactory := fakes.NewFakePhaseFactory()
+			expectedNetworkMode := "some-network-mode"
+
+			err := lifecycle.Create(context.Background(), false, false, "test", "test", "test", "test", expectedNetworkMode, fakePhaseFactory)
+			h.AssertNil(t, err)
+
+			configProvider := fakePhaseFactory.NewCalledWithProvider
+			h.AssertEq(t, configProvider.HostConfig().NetworkMode, container.NetworkMode(expectedNetworkMode))
+		})
+
+		when("clear cache", func() {
+			it("configures the phase with the expected arguments", func() {
+				verboseLifecycle := fakeLifecycle(t, true)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+
+				err := verboseLifecycle.Create(context.Background(), false, true, "test", "test", "test", "test", "test", fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertEq(t, configProvider.Name(), "creator")
+				h.AssertIncludeAllExpectedPatterns(t,
+					configProvider.ContainerConfig().Cmd,
+					[]string{"-skip-restore"},
+				)
+			})
+		})
+
+		when("clear cache is false", func() {
+			it("configures the phase with the expected arguments", func() {
+				verboseLifecycle := fakeLifecycle(t, true)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+
+				err := verboseLifecycle.Create(context.Background(), false, false, "test", "test", "test", "test", "test", fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertEq(t, configProvider.Name(), "creator")
+				h.AssertIncludeAllExpectedPatterns(t,
+					configProvider.ContainerConfig().Cmd,
+					[]string{"-cache-dir", "/cache"},
+				)
+				h.AssertSliceContains(t, configProvider.HostConfig().Binds, "/var/run/docker.sock:/var/run/docker.sock")
```

Comment:
Just being nit-picky here, I don't think the docker socket being mounted has anything to do with clear cache - it is a side-effect of passing `false` for publish. We can probably remove this line.

Reply:
You're right. I'll remove it.